src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,7 +146,7 @@ public function loadChoicesForValues(array $values, $value = null)
 
         // Optimize performance in case we have an object loader and
         // a single-field identifier
-        if (null === $value && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
+        if (!($value instanceof \Closure) && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
```

Comment:
@HeahDude It couldn't be a property path, since property paths are replaced by closures by PropertyAccessDecorator. However, it could be an array callable, hence your comment is legit.